ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -715,6 +733,34 @@ private TransactionWithMetadata transactionByHeaderAndIndex(
     return header.map(BlockHeader::getStateRoot).flatMap(worldStateArchive::getMutable);
   }
 
+  public Optional<Long> gasPrice() {
+    final long blockHeight = headBlockNumber();
+    final long[] gasCollection =
+        LongStream.range(Math.max(0, blockHeight - apiConfig.getGasPriceBlocks()), blockHeight)
+            .mapToObj(
+                l ->
+                    blockchain
+                        .getBlockByNumber(l)
+                        .map(Block::getBody)
+                        .map(BlockBody::getTransactions)
+                        .orElse(Collections.emptyList()))
```

Comment:
I think this case should be an `orElseThrow`? It doesn't represent a normal case.

Reply:
It felt like an "it could never happen situation," if we can't get blocks by number behind the head block.  Throwing an error does seem more appropriate instead of silently healing.

I supposed if we are querying the gas right when we decide to switch to a shorter but heavier chain it _might_ catch those blocks.  Simply send another request in that case.